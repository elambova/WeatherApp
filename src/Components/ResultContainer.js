import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ResultContainer extends Component {
  static propTypes = {
    data: PropTypes.object.isRequired,
  };

  render() {
    const { data } = this.props;
    const date = new Date().toDateString();

    return (
      <React.Fragment>
        <h2>{data.city} </h2>
        <h3 className="date">{date}</h3>

        <img
          src={
            data.weather.icon !== undefined
              ? `https://www.weatherbit.io/static/img/icons/${data.weather.icon}.png`
              : `http://openweathermap.org/img/wn/${data.weather[0].icon}@2x.png`
          }
          alt={`${data.weather.description}`}
        />
        <p>{data.weather.description}</p>
        <p>
          Temperature: <strong>{data.temp}</strong> &#8451;
        </p>
        <img
          id="city-pic"
          src={`${data.picture}`}
          alt={`${data.city}, ${data.country}`}
        />
        <p className="tags-image">{data.tags}</p>
      </React.Fragment>
    );
  }
}
