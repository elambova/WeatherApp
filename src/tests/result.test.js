import React from "react";
import Enzyme, { shallow, mount } from "enzyme";
import ResultContainer from "../Components/ResultContainer";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

const data = {
  city: "Sofia",
  country: "Bulgaria",
  picture:
    "https://pixabay.com/get/54e2d3474c54ad14f1dc84609629317f123cd9e4574c704c7c297ad49449c250_640.jpg",

  weather: {
    icon: "c01d",
    description: "Clear sky",
  },
};

describe("result container", () => {
  test("should be exists", () => {
    const resultContainer = shallow(<ResultContainer data={data} />);

    expect(resultContainer.exists()).toBe(true);
  });

  test("render", () => {
    const resultContainer = shallow(<ResultContainer data={data} />);

    expect(resultContainer).toMatchSnapshot();
  });

  test("renders without crashing given the required props", () => {
    const resultContainer = shallow(<ResultContainer data={data} />);

    expect(toJson(resultContainer)).toMatchSnapshot();
  });

  test("required props type Object", () => {
    const resultContainer = shallow(<ResultContainer data={data} />);

    expect(typeof toJson(resultContainer).children[0].props).toBe("object");
  });
});
