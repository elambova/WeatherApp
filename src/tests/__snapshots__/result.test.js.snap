// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`result container render 1`] = `ShallowWrapper {}`;

exports[`result container renders without crashing given the required props 1`] = `
<Fragment>
  <h2>
    Sofia
     
  </h2>
  <h3
    className="date"
  >
    Thu Aug 06 2020
  </h3>
  <img
    alt="Clear sky"
    src="https://www.weatherbit.io/static/img/icons/c01d.png"
  />
  <p>
    Clear sky
  </p>
  <p>
    Temperature: 
    <strong />
     â„ƒ
  </p>
  <img
    alt="Sofia, Bulgaria"
    id="city-pic"
    src="https://pixabay.com/get/54e2d3474c54ad14f1dc84609629317f123cd9e4574c704c7c297ad49449c250_640.jpg"
  />
  <p
    className="tags-image"
  />
</Fragment>
`;
